cluster/src/shard_lock_manager.rs

```diff
@@ -380,6 +411,93 @@ impl ShardLock {
         Bytes::from(key)
     }
 
+    /// The shard lock is allowed to acquired in a fast way if the lock has been
+    /// created by this node just before.
+    ///
+    /// The slow way has to wait for the previous lock lease expired. And the
+    /// fast way is to reuse the lease if it is not expired and the lock's value
+    /// is the same.
+    async fn maybe_fast_acquire_lock(&self, etcd_client: &mut Client) -> Result<Option<LeaseInfo>> {
+        if !self.enable_fast_reacquire {
+            return Ok(None);
+        }
+
+        let resp = etcd_client
+            .get(self.key.clone(), None)
+            .await
+            .context(GetLockKey {
+                shard_id: self.shard_id,
+            })?;
+        if resp.kvs().len() != 1 {
+            warn!(
+                "Expect exactly one key value pair, but found {} kv pairs, shard_id:{}",
+                resp.kvs().len(),
+                self.shard_id
+            );
+            return Ok(None);
+        }
+        let lease_id = resp.kvs()[0].lease();
```

Comment:
Should add a judgement about if `kvs` is empty?

Reply:
I guess the L431 has ensured it must contain exactly one element in the list.